corehq/apps/repeaters/models.py

```diff
@@ -535,13 +536,19 @@ def fire(self, force_send=False):
     def post(self, post_info, tries=0):
         tries += 1
         try:
-            response = simple_post_with_logged_timeout(
-                self.domain,
-                post_info.payload,
-                self.url,
-                headers=post_info.headers,
-                timeout=POST_TIMEOUT,
-            )
+            generator = self.get_payload_generator(self.format_or_default_format())
+            if generator.format_label == 'XML' and hasattr(self.repeater, 'operation'):
```

Comment:
Is there a reason you didn't want to explicitly set a `repeater.method`? This seems to be a bit hacky.